src/controllers/orderController.js

```diff
@@ -162,4 +164,50 @@ const deleteOrder = async (req, res) => {
   */
 };
 
-module.exports = { createOrder, getAllOrders, getSingleOrder, deleteOrder };
+const handleStripePayment = async (req) => {
+  try {
+    // Receive Stripe Signature
+    const stripeSignature = req.headers['stripe-signature'];
+
+    //  Construct the Webhook Event
+    const event = stripe.webhooks.constructEvent(
+      req.body, // Raw text body payload received from Stripe
+      stripeSignature, // Value of the `stripe-signature` header from Stripe
+      process.env.STRIPE_CLI_WEBHOOK_SECRET // Webhook Signing Secret
+    );
+
+    //  Retrieve Payment Intent and Order Information
+    const paymentIntent = event.data.object; // Payment intent information
+    const orderId = paymentIntent.metadata.orderId; // Order ID passed in metadata
+    // Additional order-related info if needed
+
+    //  Update Order Status
+    const order = await Order.findById(orderId);
+
+    if (event.type === 'payment_intent.succeeded') {
+      // Update order status to successful
+      order.orderStatus = 'payment_successful';
+    } else if (event.type === 'payment_intent.payment_failed') {
+      // Update order status to failed
+      order.orderStatus = 'payment_failed';
+    }
+
+    // Save the updated order
+    await order.save();
+
+    // Handle other actions based on the event type if needed
+
+    return { success: true };
+  } catch (error) {
+    console.error('Error handling Stripe payment:', error);
+    return { success: false, error: error.message };
```

Comment:
In a controller handler `return` returns a value to the calling backend function but it _does not send an HTTP response_. So you'll see in your terminal when running the Stripe CLI that Stripe never gets a response and after waiting around a while it will print:

```
[ERROR] Failed to POST: Post "http://localhost:8000/api/v1/orders/payment_status": context deadline exceeded (Client.Timeout exceeded while awaiting headers)
```

https://support.stripe.com/questions/webhooks-how-to-investigate-and-fix-timed-out#:~:text=Stripe%20expects%20your%20server%20to,the%20Event%20was%20successfully%20received.

We need to instead be returning a status OK (200-299 status code; in our case just a plain 200 will suffice). If you want you can return the json {success: true} but I don't think Stripe cares/checks the body anyway. 

In the error case, we could check the error more to figure out if this is a problem with what Stripe sent us (BAD\_REQUEST, or something else in the 400s) or a problem with our code (500s error). But most likely, since Stripe is always sending a specific payload, it's unlikely that they will suddenly start sending us some bad data. So we can probably just remove this try/catch; and let the error be handled by the error-handler middleware which will log the error for us already and return a 500.